internal/provider/incident_alert_route_resource_test.go

```diff
@@ -0,0 +1,492 @@
+        content = [
+          {
+            content = [
+              {
+                attrs = {
+                  label   = "Payload → Title"
+                  missing = false
+                  name    = "title"
+                }
+                type = "varSpec"
+              },
+            ]
+            type = "paragraph"
+          },
+        ]
+        type = "doc"
+      })
+    }
+    description = {
+      literal = jsonencode({
+        content = [
+          {
+            content = [
+              {
+                attrs = {
+                  label   = "Payload → Description"
+                  missing = false
+                  name    = "description"
+                }
+                type = "varSpec"
+              },
+            ]
+            type = "paragraph"
+          },
+        ]
+        type = "doc"
+      })
+    }
+    attributes  = []
+    expressions = []
+  }
+}
+
+resource "incident_alert_route" "comprehensive" {
+  name       = %[1]q
+  enabled    = true
+  is_private = false
+
+  alert_sources = [
+    {
+      alert_source_id = incident_alert_source.http_test.id
+      condition_groups = []
+    }
+  ]
+
+  condition_groups = [
+    {
+      conditions = [
+        {
+          subject   = "alert.title"
+          operation = "is_set"
+          param_bindings = []
+        }
+      ]
+    }
+  ]
+
+  expressions = []
+
+  channel_config = []
+
+  escalation_config = {
+    auto_cancel_escalations = true
+    escalation_targets = []
+  }
+
+  incident_config = {
+    auto_decline_enabled = false
+    enabled              = true
+    condition_groups     = []
+    defer_time_seconds   = 300
+    grouping_keys        = []
+  }
+
+  incident_template = {
+    custom_fields = []
+    name = {
+      autogenerated = true
+      value = {
+        literal = jsonencode({
+          content = [
+            {
+              content = [
+                {
+                  attrs = {
+                    label   = "Alert → Title"
+                    missing = false
+                    name    = "alert.title"
+                  }
+                  type = "varSpec"
+                }
+              ]
+              type = "paragraph"
+            }
+          ]
+          type = "doc"
+        })
+      }
+    }
+    summary = {
+      autogenerated = true
+      value = {
+        literal = jsonencode({
+          content = [
+            {
+              content = [
+                {
+                  attrs = {
+                    label   = "Alert → Description"
+                    missing = false
+                    name    = "alert.description"
+                  }
+                  type = "varSpec"
+                }
+              ]
+              type = "paragraph"
+            }
+          ]
+          type = "doc"
+        })
+      }
+    }
+    start_in_triage = {
+      value = {
+        literal = "true"
+      }
+    }
+    severity = {
+      merge_strategy = "first-wins"
```

Comment:
Does this make sense? Defining a strategy without a binding to apply that strategy to